docs/docs/apis/core/hd-wallet-api.md

```diff
@@ -0,0 +1,298 @@
+    System.out.println("Root key available");
+}
+
+// Get root private key bytes
+Optional<byte[]> rootPrivateKey = wallet.getRootPvtKey();
+if (rootPrivateKey.isPresent()) {
+    byte[] keyBytes = rootPrivateKey.get();
+    System.out.println("Root private key: " + keyBytes.length + " bytes");
+}
+
+// Get stake address for delegation
+String stakeAddress = wallet.getStakeAddress();
+System.out.println("Stake Address: " + stakeAddress);
+```
+
+### Transaction Signing
+
+Sign transactions with wallet UTXOs:
+
+```java
+// Sign transaction with specific UTXOs
+Set<WalletUtxo> walletUtxos = getWalletUtxos(); // Your UTXO collection logic
+Transaction signedTx = wallet.signTransaction(transaction, walletUtxos);
+
+// Sign with stake key
+Transaction stakeSignedTx = wallet.signWithStakeKey(transaction);
+
+// Combined signing
+Transaction fullySignedTx = wallet.signTransaction(
+    wallet.signWithStakeKey(transaction), 
+    walletUtxos
+);
+```
+
+## Best Practices
+
+### Secure Wallet Creation
+
+```java
+// Generate wallet securely
+Wallet secureWallet = Wallet.create(Networks.mainnet());
+String mnemonic = secureWallet.getMnemonic();
+
+// Store mnemonic securely (encrypted storage recommended)
+// Never store mnemonic in plain text
+storeSecurely(mnemonic);
+
+// Recreate wallet when needed
+Wallet restoredWallet = Wallet.createFromMnemonic(Networks.mainnet(), mnemonic);
+```
+
+### Address Management Strategy
+
+```java
+// Organize addresses by purpose
+public class WalletManager {
+    private final Wallet wallet;
+    
+    public WalletManager(Network network, String mnemonic) {
+        this.wallet = Wallet.createFromMnemonic(network, mnemonic);
+    }
+    
+    // Receiving addresses (account 0)
+    public Address getReceivingAddress(int index) {
+        return wallet.getBaseAddress(0, index);
+    }
+    
+    // Change addresses (account 1)
+    public Address getChangeAddress(int index) {
+        return wallet.getBaseAddress(1, index);
+    }
+    
+    // Business addresses (account 2)
+    public Address getBusinessAddress(int index) {
+        return wallet.getBaseAddress(2, index);
+    }
+}
+```
+
+## HD Wallet vs Account API Comparison
+
+| Feature | HD Wallet API | Account API |
+|---------|---------------|-------------|
+| **Key Management** | Hierarchical key derivation | Single key pair |
+| **Address Generation** | Multiple derived addresses | Fixed addresses |
+| **Privacy** | High (new address per transaction) | Limited |
+| **Backup** | Single mnemonic | Individual keys |
+| **Use Case** | Privacy-focused applications | Simple applications |
+| **Complexity** | Medium to High | Low |
+| **Standards** | BIP32/CIP1852 compliant | Basic |
+| **Multi-Account** | Native support | Manual management |
+
+Choose HD Wallet API for applications requiring enhanced privacy, multiple addresses, or hierarchical key management. Use Account API for simpler use cases where basic account functionality is sufficient.
```

Comment:
Instead of 

`Use Account API for simpler use cases where basic account functionality is sufficient.` . 

We can say

`Use Account API if your application is using only one address or account`.